src/main/rx/cc2500_frsky_x.c

```diff
@@ -343,89 +358,78 @@ rx_spi_received_e frSkyXHandlePacket(uint8_t * const packet, uint8_t * const pro
+                    memcpy(&telemetryRxBuffer[remoteNewPacketId].data, &packet[22], TELEMETRY_FRAME_SIZE);
+                    telemetryRxBuffer[remoteNewPacketId].needsProcessing = true;
 
-                                        break;
-                                    }
+                    responseToSend.raw = 0;
+                    uint8_t remoteToAckId = (remoteAckId + 1) % TELEMETRY_SEQUENCE_LENGTH;
+                    if (remoteNewPacketId != remoteToAckId) {
+                        while (remoteToAckId != remoteNewPacketId) {
+                            if (!telemetryRxBuffer[remoteToAckId].needsProcessing) {
+                                responseToSend.data.ackSequenceId = remoteToAckId;
+                                responseToSend.data.retransmissionRequested = 1;
 
-                                    remoteToAckId = (remoteToAckId + 1) % TELEMETRY_SEQUENCE_LENGTH;
-                                }
-                            }
+                                receiveTelemetryRetryCount++;
 
-                            if (!responseToSend.data.retransmissionRequested) {
-                                receiveTelemetryRetryCount = 0;
-
-                                remoteToAckId = (remoteAckId + 1) % TELEMETRY_SEQUENCE_LENGTH;
-                                uint8_t remoteNextAckId;
-                                while (telemetryRxBuffer[remoteToAckId].needsProcessing && remoteToAckId != remoteAckId) {
-                                    remoteNextAckId = remoteToAckId;
-                                    remoteToAckId = (remoteToAckId + 1) % TELEMETRY_SEQUENCE_LENGTH;
-                                }
-                                remoteAckId = remoteNextAckId;
-                                responseToSend.data.ackSequenceId = remoteAckId;
+                                break;
                             }
 
-                            if (receiveTelemetryRetryCount >= 5) {
-                                 remoteProcessedId =  TELEMETRY_SEQUENCE_LENGTH - 1;
-                                 remoteAckId =  TELEMETRY_SEQUENCE_LENGTH - 1;
-                                 for (unsigned i = 0; i < TELEMETRY_SEQUENCE_LENGTH; i++) {
-                                     telemetryRxBuffer[i].needsProcessing = false;
-                                 }
+                            remoteToAckId = (remoteToAckId + 1) % TELEMETRY_SEQUENCE_LENGTH;
+                        }
+                    }
 
-                                 receiveTelemetryRetryCount = 0;
-                             }
+                    if (!responseToSend.data.retransmissionRequested) {
+                        receiveTelemetryRetryCount = 0;
 
-                            packetTimerUs = micros();
-                            frameReceived = true; // no need to process frame again.
+                        remoteToAckId = (remoteAckId + 1) % TELEMETRY_SEQUENCE_LENGTH;
+                        uint8_t remoteNextAckId;
```

Comment:
If the while loop below does not execute, you would be using uninitialised ```remoteNextAckId``` in ```remoteAckId = remoteNextAckId```.

Reply:
I'm not sure how to fix this as it is not my code, I only changed indentation here. I'll take a look as it might be a possible bug. Good catch!

Reply:
Looks like `uint8_t remoteNextAckId = remoteToAckId` here will fix the bug.

Reply:
OK, bear with me, I'll have to test this before pushing.